update-agent/dbus/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+//! Query the current update status:
+//! ```bash
+//! gdbus call --session -d org.worldcoin.UpdateAgentManager1 -o \
+//! '/org/worldcoin/UpdateAgentManager1' -m \
+//! org.freedesktop.DBus.Properties.Get org.worldcoin.UpdateAgentManager1 Progress
+//! ```
+//!
+//! Monitor for signals:
+//! ```bash
+//! export DBUS_SESSION_BUS_ADDRESS=unix:path=/tmp/worldcoin_bus_socket
+//! dbus-monitor --session type='signal',sender='org.worldcoin.UpdateAgentManager1'
+//! ```
+
+use serde::{Deserialize, Serialize};
+use zbus::interface;
+use zbus::zvariant::{OwnedValue, Type, Value};
+
+/// A trait representing update progress behavior.
+///
+/// This trait is implemented by types that can provide information about the current update status.
```

Comment:
maybe explain more clearly why the trait and wrapper type exist. Its to allow dependency injection, mocking, and sharing the same types across both client and server.